code/modules/mob/living/silicon/silicon_defense.dm

```diff
@@ -83,12 +92,27 @@
 			src.take_bodypart_damage(10)
 	src << "<span class='userdanger'>*BZZZT*</span>"
 	src << "<span class='danger'>Warning: Electromagnetic pulse detected.</span>"
+	for(var/mob/living/M in buckled_mobs)
+		if(prob(severity*50))
+			unbuckle_mob(M)
+			M.Weaken(2)
+			M.visible_message("<span class='boldwarning'>[M] is thrown off of [src]!</span>")
 	flash_act(affect_silicon = 1)
 	..()
 
 /mob/living/silicon/bullet_act(obj/item/projectile/Proj)
 	if((Proj.damage_type == BRUTE || Proj.damage_type == BURN))
 		adjustBruteLoss(Proj.damage)
+		if(prob(Proj.damage*1.5))
+			for(var/mob/living/M in buckled_mobs)
+				M.visible_message("<span class='boldwarning'>[M] is knocked off of [src]!</span>")
+				unbuckle_mob(M)
+				M.Weaken(2)
+	if(Proj.stun || Proj.weaken)
+		for(var/mob/living/M in buckled_mobs)
+			M.Stun(Proj.stun)
+			M.Weaken(Proj.weaken)
+			M.visible_message("<span class='boldwarning'>[M] is stunned by the [Proj]!</span>")
```

Comment:
BAD, BAD, pass the projectile's on_hit to one of the buckled mobs instead, don't hit multiple things with it